### PR TITLE
[otbn] Test access to randomness

### PR DESCRIPTION
**Pull request by imphil**
Test access to randomness from OTBN software. The test is triggered from Ibex to enable it to configure the EDN appropriately first.

~Only the last patch in this PR is relevant; the other patches will go away with the merge of #7341.~

### CODE REVIEW

File: sw/device/tests/otbn/otbn_randomness_test.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/otbn.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+// Temporary solution to configue/enable the EDN and CSRNG to allow OTBN to run
```
Comment (rswarbrick) [inline sw/device/tests/otbn/otbn_randomness_test.c:14]: Nit: configu**r**e

File: test/systemtest/config.py [hunk 2]
```diff
@@ -33,6 +33,11 @@
         "verilator_extra_args": ['+OTBN_USE_MODEL=1'],
         "targets": ["sim_verilator"],
     },
+    {
+        "name": "otbn_randomness_test",
+        # TODO: Run this test also against the ISS URND and RND are implemented.
```
Comment (rswarbrick) [inline test/systemtest/config.py:38]: Nit: "once URND and RND are implemented"?

**Review by rswarbrick — APPROVED** [review]
Looks good! I contend that two equal values shouldn't happen before tape-out, even at the 32-bit size but I *guess* it's possible :-)

File: sw/device/tests/otbn/otbn_randomness_test.c (hunk 1 above)
Comment (imphil) [reply sw/device/tests/otbn/otbn_randomness_test.c]: Copy & paste :-) Fixed everywhere.

**Pull request merged**